cmd/nerdctl/main.go

```diff
@@ -450,3 +454,14 @@ func AddPersistentStringArrayFlag(cmd *cobra.Command, name string, aliases, nonP
 		}
 	}
 }
+
+func checkExperimental(cmd *cobra.Command, args []string) error {
+	experimental, err := cmd.Flags().GetBool("experimental")
+	if err != nil {
+		return err
+	}
+
+	if !experimental {
+		return fmt.Errorf("this is experimental feature, you should enable experimental config")
```

Comment:
"this" is unclear. Probably better to remove `checkExperimental()` and add the checks with detailed messages in the each of the experimental commands

Reply:
```go
func checkExperimental(experimental string) func(cmd *cobra.Command, args []string) error {
	return func(cmd *cobra.Command, args []string) error {
		experimental, err := cmd.Flags().GetBool("experimental")
		if err != nil {
			return err
		}
		if !experimental {
			return fmt.Errorf("%s is experimental feature, you should enable experimental config", experimental)
		}
		return nil
	}
}
```

wrap the `PreRunE`

Reply:
SGTM